evaluator/package_cache.go

```diff
@@ -0,0 +1,139 @@
+package evaluator
+
+import (
+	"app/base/database"
+	"app/base/utils"
+	"sync"
+)
+
+var memoryPackageCache *PackageCache
+
+type PackageCacheMetadata struct {
+	ID              int
+	NameID          int
+	Name            string
+	Evra            string
+	SummaryHash     []byte
+	DescriptionHash []byte
+}
+
+type PackageCache struct {
+	mtx          sync.RWMutex
+	data         []PackageCacheMetadata
+	byID         map[int]*PackageCacheMetadata
+	byNevra      map[string]*PackageCacheMetadata
+	latestByName map[string]*PackageCacheMetadata
+	nameByID     map[int]string
+}
+
+func NewPackageCache() *PackageCache {
+	c := new(PackageCache)
+	c.data = []PackageCacheMetadata{}
+	c.byID = map[int]*PackageCacheMetadata{}
+	c.byNevra = map[string]*PackageCacheMetadata{}
+	c.latestByName = map[string]*PackageCacheMetadata{}
+	c.nameByID = map[int]string{}
+	return c
+}
+
+func (c *PackageCache) load() {
```

Comment:
As it's called from outside of this file, I would call it `Load()`

Reply:
renamed